Modules/ThirdParty/GDCM/src/gdcm/Source/Common/gdcmSwapper.txx

```diff
@@ -106,7 +106,9 @@ namespace gdcm
     }
   template <> inline float SwapperNoOp::Swap<float>(float val)
     {
-    return Swap((uint32_t)val);
+    uint32_t temp;
+    memcpy(&temp, &val, sizeof(uint32_t));
+    return static_cast<float>(Swap(temp));
```

Comment:
IMHO, not yet sure about it, **maybe i am wrong**, is this cast to float from uint32 OK?
Is it not required to copy bits back to float instead?
```cpp
   uint32_t temp;
   memcpy(&temp, &val, 4);
   temp = Swap(temp);
   memcpy(&val, &temp, 4);
   return val;
```

CC @N-Dekker, could you kindly take a look?

Reply:
@issakomi Thanks for asking me, but honestly I don't know what `SwapperNoOp::Swap<float>` is supposed to do. Does `NoOp` mean "do nothing", as in https://en.wikipedia.org/wiki/NOP_(code) ?

Quite recently (honestly) I became aware that a `union` should never be used to convert one type to another. It would cause _undefined behavior_. If such a bit-wise conversion is really necessary (as in this case), it can be done more properly by `std::memcpy`. (Or by `std::bit_cast`, if you're using C++20 : https://en.cppreference.com/w/cpp/numeric/bit_cast ) 

My two cents!

Reply:
> Does NoOp mean "do nothing"

Looks like it depends on _GDCM_WORDS_BIGENDIAN_

```cpp
#ifdef GDCM_WORDS_BIGENDIAN
class SwapperDoOp
{
public:
  template <typename T> static T Swap(T val) {return val;}
  template <typename T> static void SwapArray(T *, unsigned int ) {}
};
...
#else
class SwapperNoOp
{
public:
  template <typename T> static T Swap(T val) {return val;}
  template <typename T> static void SwapArray(T *, size_t ) {}
};
...
```

Yes, that solution with union is know as being _not always working_.

Thank you for the answer!
Maybe i look at it later again, actually i got a headache already. :)

Reply:
> actually i got a headache already. :)

Sorry to hear so @issakomi ! Stay away from the computer for a little while, and I'm sure you will soon get well again  :)

Reply:
@issakomi thanks for your review... I'll take another look...

Reply:
@issakomi you're right, my patch was wrong, I'll tag you in my fix to upstream GDCM...

Reply:
Thank you. If fact i don't have a way to validate just now with some data set or GDCM example. As you find it right.

Reply:
BTW I've been playing with an old PowerPC G5 Mac running OpenBSD, which is how I've come across these issues...